esp/services/ldapenvironment/ldapenvironmentService.cpp

```diff
@@ -87,61 +87,77 @@ bool CldapenvironmentEx::onLDAPQueryDefaults(IEspContext &context, IEspLDAPQuery
     resp.setSharedUsersBaseDN(sharedUsersBaseDN.str());
     resp.setSharedResourcesBaseDN(sharedResourcesBaseDN.str());
     resp.setSharedWorkunitsBaseDN(sharedWorkunitsBaseDN.str());
-    resp.setRetcode(0);
     return true;
 }
 
-const char * CldapenvironmentEx::formatOUname(StringBuffer &ou, const char * envName, int mode, const char * sharedOU, const char * reqBaseDN, const char * privateOU)
+const char * CldapenvironmentEx::formatOUname(StringBuffer &ou, const char * envName, int mode, const char * sharedOU, const char * reqBaseDN, const char * standAloneOU)
 {
     switch (mode)
     {
-        case 0: // Generated Private OU
-            ou.appendf("%s,ou=%s,ou=%s", privateOU, envName, ldapRootOU.str());
+        case COUMode_GenerateStandAlone: // Generated Stand Alone OU
+            ou.appendf("%s,ou=%s,%s", standAloneOU, envName, ldapRootOU.str());
             break;
-        case 1: // Global/Shared OU
+        case COUMode_UseGlobal: // Global/Shared OU
             ou.append(sharedOU);//specified in ldapconfiguration
             break;
-        case 2: // Custom User-provided OU
+        case COUMode_CreateCustom: // Custom User-provided OU
             ou.append(reqBaseDN);
             break;
     }
     return ou.str();
 }
 
+const char * CldapenvironmentEx::generatePassword(StringBuffer &pwd, int len)
+{
+    const char alphaUC[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    const char alphaLC[] = "abcdefghijklmnopqrstuvwxyz";
+    const char numeric[] = "0123456789";
+    const char symbol[] = "~`!@#$%^&*()_-+={[}]|:;<,>.?/";
+
+    srand(time(0));
+    for(int i = 0; i < len; i += 4)
+        pwd.appendf("%c", alphaUC[rand() %  (sizeof(alphaUC)-1) ]);
+    for(int i = 1; i < len; i += 4)
+        pwd.appendf("%c", alphaLC[rand() %  (sizeof(alphaLC)-1) ]);
```

Comment:
Did you try a simple append?  It should use the character version of append.

`pwd.append(alphaLC[rand() %  (sizeof(alphaLC)-1) ]);`

Reply:
You are correct